### PR TITLE
Add schema id translation

### PR DESCRIPTION
**Pull request by mihaitodor**
~This requires merging #2918 first.~ Sorted!

TODO:
- [x] Unit tests
- [x] Integration tests

### CODE REVIEW

File: internal/impl/confluent/sr/serde.go
Comment (rockwotj) [inline internal/impl/confluent/sr/serde.go:40]: off by one :)

Comment (mihaitodor) [reply internal/impl/confluent/sr/serde.go]: Well spotted, thank you! ❤️ I fixed it now.

File: internal/impl/confluent/sr/client.go
Comment (mihaitodor) [inline internal/impl/confluent/sr/client.go:313]: Note that the `github.com/twmb/franz-go/pkg/sr` client doesn't do 3 retries under the hood. I can try to shoehorn this behaviour into the custom `roundTripper` if you think it's important to preserve it.

File: internal/impl/confluent/sr/client.go
Comment (rockwotj) [inline internal/impl/confluent/sr/client.go:47]: Is Travis up for allowing some middleware option that intercepts the http request properly?

File: internal/impl/kafka/enterprise/schema_registry_input.go
Comment (rockwotj) [inline internal/impl/kafka/enterprise/schema_registry_input.go:164]: instead of a raw string can we use some typed wrapper so it's not possible for another component to clobber this? Then we don't need to do they empty string special case.

File: internal/impl/kafka/enterprise/schema_registry_output.go
Comment (rockwotj) [inline internal/impl/kafka/enterprise/schema_registry_output.go:350]: Are we waiting on this before we merge?

File: internal/impl/confluent/sr/client.go
Comment (mihaitodor) [reply internal/impl/confluent/sr/client.go:47]: I did ask, but he hasn't gotten around to it yet... Not sure if we should block this PR until that happens, but yeah, it would be great to avoid doing this.

File: internal/impl/kafka/enterprise/schema_registry_output.go
Comment (mihaitodor) [reply internal/impl/kafka/enterprise/schema_registry_output.go:350]: I think it's fine to leave it as a TODO for now. The main use case is to do the ID translation.

**Review by rockwotj — APPROVED** [review]
outside of my comment about the typed key in resources this LGTM

File: CHANGELOG.md
Comment (Jeffail) [inline CHANGELOG.md:31]: Is this correctly placed under v4.39?

File: internal/impl/confluent/sr/client.go
Comment (Jeffail) [reply internal/impl/confluent/sr/client.go:313]: Nah I think this is fine, the concept of attempts is abstract and users don't have any existing documentation or control so I think it's okay to change the behaviour slightly here.

File: CHANGELOG.md
Comment (mihaitodor) [reply CHANGELOG.md]: Oops, thank you! ❤️

**Pull request merged**
